src/get-glamor-classname.js

```diff
@@ -25,31 +25,32 @@ function extractGlamorStyles(className = '') {
 
 export default getGlamorClassName
 
-function getGlamorClassName(styles, props, cssOverrides, theme, context) {
+function getGlamorClassName({
+  styles,
+  props,
+  cssOverrides,
+  cssProp,
+  theme,
+  context,
+}) {
+  const {
+    glamorStyles: parentGlamorStyles,
+    glamorlessClassName,
+  } = extractGlamorStyles(props.className)
   const {mappedArgs, nonGlamorClassNames} = handleStyles(
-    styles,
+    [...styles, parentGlamorStyles, cssOverrides, cssProp],
```

Comment:
Is this determining precedence? If so, this is the order you expect them in? Are they reduced? I would expect styles to have highest importance.